src/compiler/checker.ts

```diff
@@ -5864,9 +5864,9 @@ namespace ts {
                 for (const declaration of symbol.declarations) {
                     if (declaration.kind === SyntaxKind.EnumDeclaration) {
                         for (const member of (<EnumDeclaration>declaration).members) {
-                            const memberType = getLiteralType(getEnumMemberValue(member)!, enumCount, getSymbolOfNode(member)); // TODO: GH#18217
+                            const memberType = getFreshTypeOfLiteralType(getLiteralType(getEnumMemberValue(member)!, enumCount, getSymbolOfNode(member))); // TODO: GH#18217

```

Comment:
I'm sort of surprised you're storing the fresh type in the symbol instead of the regular type. I would expect to see the regular type in the symbol and then freshness added by `getTypeOfSymbol`.